### PR TITLE
[hobbleabbas] Synthetic tests

### PR DESCRIPTION
**Pull request by dash-xa**

### CODE REVIEW

File: repos/seaborn
Comment (dash-xa) [inline repos/seaborn:1]: Can you remove? We are cloning repos dynamically now

File: benchmarking/trajectories/34d38b.traj
Comment (dash-xa) [inline benchmarking/trajectories/34d38b.traj:1]: Should remove, no need to have `trajectories/` in repo

Comment (hobbleabbas) [reply benchmarking/trajectories/34d38b.traj:1]: yep

File: benchmarking/generate_solution.py [hunk 1]
```diff
@@ -24,67 +23,6 @@
 logger = logging.getLogger(__name__)
 
 
-@dataclass(frozen=True)
-class ActionsArguments(FlattenedAccess, FrozenSerializable):
```
Comment (dash-xa) [inline benchmarking/generate_solution.py:28]: We should keep these here, they are copied from SWE-agent and only specific to the miner. Will make copy/pasting into subnet repo easier

File: benchmarking/grade_output.py [hunk 2]
```diff
@@ -1,39 +1,269 @@
...
+    "seaborn": {
+        "path": "tests",
+        "label_side": "left",
+        "framework": "pytest"
+    }
+}
+
+SUPPORTED_TEST_FRAMEWORKS = ["pytest"]
+
+def create_script_arguments(model_name: str, repo_path: Path) -> ScriptArguments:
```
Comment (dash-xa) [inline benchmarking/grade_output.py:25]: Can we just reuse the same `create_script_arguments` func from `generate_solution.py`?

File: benchmarking/grade_output.py [hunk 3]
```diff
@@ -1,39 +1,269 @@
...
     
-    return DYNAMIC_CHECKLIST_WEIGHT * statistics.mean(vars(miner_output_score.dynamic_checklist_scores).values()) + \
-        ADDRESSES_PROBLEM_WEIGHT * miner_output_score.addresses_problem_in_statement + \
-        LOGICAL_SOLUTION_WEIGHT * miner_output_score.logical_solution + \
-        BREVITY_WEIGHT * miner_output_score.brevity_and_cleanliness_of_code + \
-        POTENTIAL_BUGS_WEIGHT * miner_output_score.potential_bugs_generated
+    print("Generated Synthetic Test: ", synthetic_test)
+
+    # Validate with pytest before creating a permanent test file
+    with tempfile.NamedTemporaryFile(mode='w', suffix='.py') as tmp:
```
Comment (dash-xa) [inline benchmarking/grade_output.py:206]: nice

File: benchmarking/grade_output.py [hunk 4]
```diff
@@ -44,79 +274,56 @@ def grade_miner_solution(
...
+    # print("Generated and wrote synthetic test: ")
+    # print(synthetic_test)
+
+    # repo_path = Path(__file__).parent.parent / "repos" / "seaborn"
+    # run_tests_for_miner_solution(repo_path=repo_path, patch="")
 
-def run_test_patch(repo_path: str) -> None:
-    # Spin up a container with the repo, with the patch injected
-    # Run the tests before and after
-    # Return the results
```
Comment (dash-xa) [inline benchmarking/grade_output.py:121]: Delete?

**Pull request closed without merging**